src/imitation/scripts/common/demonstrations.py

```diff
@@ -27,25 +31,45 @@ def fast():
     n_expert_demos = 1  # noqa: F841
 
 
-def guess_expert_dir(data_dir: str, env_name: str) -> str:
-    rollout_hint = env_name.rsplit("-", 1)[0].replace("/", "_").lower()
-    return os.path.join(data_dir, "expert_models", f"{rollout_hint}_0")
-
-
-@demonstrations_ingredient.config_hook
-def hook(config, command_name, logger):
-    """If rollout_path not set explicitly, then guess it based on environment name."""
-    del command_name, logger
-    updates = {}
-    if config["demonstrations"]["rollout_path"] is None:
-        data_dir = config["demonstrations"]["data_dir"]
-        env_name = config["common"]["env_name"].replace("/", "_")
-        updates["rollout_path"] = os.path.join(
-            guess_expert_dir(data_dir, env_name),
-            "rollouts",
-            "final.pkl",
-        )
-    return updates
+@demonstrations_ingredient.capture
+def get_expert_trajectories(
+    rollout_path: str,
+) -> Sequence[types.Trajectory]:
+    if rollout_path is not None:
+        return load_expert_trajs()
+    else:
+        return generate_expert_trajs()
+
+
+@demonstrations_ingredient.capture
+def generate_expert_trajs(
+    n_expert_demos: Optional[int],
+    common,
+) -> Optional[Sequence[types.Trajectory]]:
+    """Generates expert demonstrations.
+
+    Args:
+        n_expert_demos: The number of trajectories to load.
+            Dataset is truncated to this length if specified.
+        common: The common config.
+
+    Returns:
+        The expert trajectories.
+
+    Raises:
+        ValueError: If n_expert_demos is None.
+    """
+    if n_expert_demos is None:
+        raise ValueError("n_expert_demos must be specified when rollout_path is None")
+
+    rollout_env = DummyVecEnv(
+        [lambda: RolloutInfoWrapper(gym.make(common["env_name"])) for _ in range(4)],
```

Comment:
I think the rationale was to keep all modules as self contained as possible. Since generating the vec-env is just a 1-liner here I decided not to use any of the utility functions. 
I will have a look at the two utils you mentioned. But tbh just the fact that we have to similar utils scattered in different modules makes me feel uneasy.

Reply:
My point is done right the environment creation *isn't* a one-liner. This doesn't seed the environments, so they could all end up producing identical rollouts. Better to use a utility method that handles subtleties like this.

`util.make_vec_env` is a programmatic API that can be used from any Python code. `common.make_venv` is a Sacred capture function designed to only be called from scripts that use that ingredient, and is a light wrapper around `util.make_vec_env`. IMO it seems reasonable to have both, though happy to consider refactors.

Reply:
Good point. I used `common.make_venv` now.